addon/-private/class/modifier.ts

```diff
@@ -24,6 +30,21 @@ function deprecateForDestroyables<S>(
   );
 }
 
+/** @internal */
+export const _implementsModify = <S>(
+  instance: ClassBasedModifier<S>
+): boolean => instance.modify !== ClassBasedModifier.prototype.modify;
```

Comment:
I know I mentioned doing it this way, but I think we should actually eagerly capture `ClassBasedModifier.prototype.modify`. If we don't then it's possible for someone to haxxxors around our check here by just modifying the prototype.

---

Note: I'm 100% ok with pushback here, maybe we just don't care if you are going to do that much dastardly haxxory...

Reply:
Honestly, yeah, that’s my take. If someone does that and they get broken by it, that’s entirely their own fault. You’d have to be actively trying to haxx0rs around the deprecation (and: why? It wouldn’t actually gain you anything!) so on your own head be it at that point.